src/engine/test/source/hlp2/kvmap_test.cpp

```diff
@@ -167,11 +167,56 @@ TEST(KVParser, parser)
             fn(R"({"pure_letters":"abcdefghijklmnopqrstuvwxyz","integer":1234567890,"double":12345.67890,"mixed_string_a":"1234abcde","mixed_string_b":"1234.567890abcde"})"),
             strlen("pure_letters=abcdefghijklmnopqrstuvwxyz integer=1234567890 "
                    "double=12345.67890 mixed_string_a=1234abcde "
-                   "mixed_string_b=1234.567890abcde")}};
-
+                   "mixed_string_b=1234.567890abcde")},
+        TestCase {R"(key1=value1 key2=value2 key3="")",
+                  true,
+                  {},
+                  Options {"=", " ", "\"", "'"},
+                  fn(R"({"key1":"value1","key2":"value2","key3":null})"),
+                  31},
+        TestCase {R"(key1=value1 key2="" key3=value3)",
+                  true,
+                  {},
+                  Options {"=", " ", "\"", "'"},
+                  fn(R"({"key1":"value1","key2":null,"key3":"value3"})"),
+                  31},
+        TestCase {R"(key1=value1 key2=value2 key3=)",
+                  true,
+                  {},
+                  Options {"=", " ", "\"", "'"},
+                  fn(R"({"key1":"value1","key2":"value2","key3":null})"),
+                  29},
+        TestCase {R"(key1=value1 key2= key3=value3)",
+                  true,
+                  {},
+                  Options {"=", " ", "\"", "'"},
+                  fn(R"({"key1":"value1","key2":null,"key3":"value3"})"),
+                  29},
+        TestCase {R"(key1="value1" key2="123")",
+                  true,
+                  {},
+                  Options {"=", " ", "\"", "'"},
+                  fn(R"({"key1":"value1","key2":"123"})"),
+                  24},
+        TestCase {R"(key1="123" key2=456)",
+                  true,
+                  {},
+                  Options {"=", " ", "\"", "'"},
+                  fn(R"({"key1":"123","key2":456})"),
+                  19},
+        TestCase {R"(key1='value=1',key2=value''2,key3='value,3',key4='value=,''4')",
+                  false,
+                  {},
+                  Options {"=", ",", "'", "'"},
+                  fn(R"({"key1":"value=1","key2":"value'2","key3":"value,3","key4":"value=,'4"})"),
+                  61}
+        };
     for (auto t : testCases)
     {
         runTest(t, hlp::getKVParser);
+        runTest(t, hlp::getKVParser, "header", "");
+        runTest(t, hlp::getKVParser, "header", "tail");
+        runTest(t, hlp::getKVParser, "", "tail");
```

Comment:
This could be moved inside runTest(), right? calling it with both new parameters and testing this 3 scenarios inside that function:
1. without header and tail
2. with both
3. with header
4. with tail